source/octf/utils/container/IIteratorConst.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright(c) 2012-2018 Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause-Clear
+ */
+
+#ifndef SOURCE_OCTF_UTILS_CONTAINER_IITERATORCONST_H
+#define SOURCE_OCTF_UTILS_CONTAINER_IITERATORCONST_H
+
+#include <memory>
+
+namespace octf {
+
+/**
+ * @ingroup Containers
+ * @brief Basic const iterator interface
+ *
+ * This interface defines required methods for iterator which allows to iterate
+ * over const container
+ *
+ * @tparam type Type of container's element
+ * @tparam iterator Type of container's const iterator
+ * @note iterator must inherits by IIteratorConst
+ * @interface IIteratorConst
+ */
+template <typename type, typename const_iterator>
+class IIteratorConst {
+public:
+    IIteratorConst() = default;
+    virtual ~IIteratorConst() = default;
+
+    /**
+     * @brief Const reference operator to contaier's element at current iterator
+     * position
+     *
+     * @return Const reference to contaier's element at current iterator
+     */
+    virtual const type &operator*() const = 0;
+
+    /**
+     * @brief Const pointer operator to contaier's element at current iterator
+     * position
+     *
+     * @return Const pointer to contaier's element at current iterator
+     */
+    virtual const type *operator->() const = 0;
+
+    /**
+     * @brief Post-increment iterator operator
```

Comment:
I think this is a pre-increment operator. Post-increment operators take int as a dummy argument. Please validate all operators.